packages/flutter_tools/lib/src/linux/build_linux.dart

```diff
@@ -29,12 +29,14 @@ final RegExp errorMatcher = RegExp(r'(?:(?:.*:\d+:\d+|clang):\s)?(fatal\s)?(?:er
 Future<void> buildLinux(
   LinuxProject linuxProject,
   BuildInfo buildInfo, {
-    String target = 'lib/main.dart',
+    String? target,
     SizeAnalyzer? sizeAnalyzer,
     bool needCrossBuild = false,
-    TargetPlatform targetPlatform = TargetPlatform.linux_x64,
+    TargetPlatform? targetPlatform,
     String targetSysroot = '/',
   }) async {
+  target ??= 'lib/main.dart';
```

Comment:
These two can be moved to the initializer list (I'm guessing these are not parameter defaults because the super-class needs to be nullable?)

Reply:
For `target` this was null because the `mainPath` passed into `desktop_device` `buildForDevice` was nullable (target and mainPath are used interchangeably in this callstack). 
 But looking at it again, I think it should actually be required, I don't see a case where it's null.

https://github.com/flutter/flutter/blob/41167c957d60a2e8ece73d16f384abd740fbce2b/packages/flutter_tools/lib/src/desktop_device.dart#L204-L208 

The `targetPlatform` should also be required, I missed that the nullable `linux_device` `_targetPlatform` should actually become a nonnull `late final`. Thanks for catching, I should have noticed how weird this was (I thought it was an artifact of null unsafe code calling into this null safe code and needing more guards, but it isn't).

Reply:
Actually, g3 disagrees re: `mainPath`
```
      await device.startApp(
        app,
        // `required` argument for `startApp` but is not used.
        debuggingOptions: DebuggingOptions.disabled(BuildInfo.debug),
      );
```
https://source.corp.google.com/piper///depot/google3/mobile/flutter/testing/driver_test/lib/orchestrator.dart;l=117?q=orchestrator.dart&sq=package:piper%20file:%2F%2Fdepot%2Fgoogle3%20-file:google3%2Fexperimental

Reply:
It can't be put in an initializer list, this isn't a constructor.

I think this needs to stay as-is unfortunately.  We want callers to be able to omit `target` to default to `lib/main.dart` (as is done in g3).  However, after that point in the method we want it to be nonnull.

Setting the default parameter value:
```
String? target = 'lib/main.dart',
```
Means that a `String?` can be passed in (as opposed to the parameter omitted), and it won't be set to `'lib/main.dart'` and will be null.  However, if we `target ??= 'lib/main.dart';` then the rest of the method correctly treats `target` as if it's nonnull.

Reply:
SGTM